src/trilateration/trilateration.cpp

```diff
@@ -0,0 +1,186 @@
+ *       following two links. Please refer to them for all questions relating
+ *       to mathematic operations. All variables have been named after the
+ *       variables defined in the derivation for clarity.
+ *          http://robosub.eecs.wsu.edu/wiki/ee/hydrophones/trilateration/start
+ *          http://robosub.eecs.wsu.edu/wiki/cs/hydrophones/start
+ */
+
+#include "robosub/HydrophoneDeltas.h"
+#include "robosub/Position.h"
+#include "ros/ros.h"
+
+ros::Publisher position_pub;
+
+/*
+ * The speed of sound in water is 1484 m/s.
+ */
+static constexpr double speed_sound_in_water = 1484;
+
+/*
+ * The hydrophone position arguments. Delta is the position of the hx
+ * hydrophone along the x axis, epsilon is hy along the y axis, and zeta is hz
+ * along the z axis.
+ */
+double delta;
+double epsilon;
+double zeta;
+
+/**
+ * Callback funtion for hydrophone time delta message.
+ *
+ * @param msg A pointer to the received message.
+ *
+ * @return None.
+ */
+void deltaCallback(const robosub::HydrophoneDeltas::ConstPtr& msg)
+{
+    /*
+     * The inverting multiplication (by -1) is used to invert the time stamp
+     * delta calculation. This was a solution by Brian Moore and Ryan Summers
+     * after debug testing.
+     */
+    const double delta_x = msg->xDelta.toSec() * speed_sound_in_water * -1;
+    const double delta_y = msg->yDelta.toSec() * speed_sound_in_water * -1;
+    const double delta_z = msg->zDelta.toSec() * speed_sound_in_water * -1;
+
+    /*
+     * Calculate the distinct x, y, and z portions of the quadratic
+     * coefficients and combine them into a complete quadratic equation.
+     */
+    const double a_x = pow((delta_x / delta), 2);
+    const double a_y = pow((delta_y / epsilon), 2);
+    const double a_z = pow((delta_z / zeta), 2);
+
+    const double b_x = delta_x / pow(delta, 2) * (pow(delta, 2) -
+            pow(delta_x, 2));
+    const double b_y = delta_y / pow(epsilon, 2) * (pow(epsilon, 2) -
+            pow(delta_y, 2));
+    const double b_z = delta_z / pow(zeta, 2) * (pow(zeta, 2) -
+            pow(delta_z, 2));
+
+    const double c_x = pow((pow(delta_x, 2) - pow(delta, 2)) / (2 * delta), 2);
+    const double c_y = pow((pow(delta_y, 2) - pow(epsilon, 2)) / (2 * epsilon),
+            2);
+    const double c_z = pow((pow(delta_z, 2) - pow(zeta, 2)) / (2 * zeta), 2);
+
+    const double a = a_x + a_y + a_z - 1;
+    const double b = b_x + b_y + b_z;
+    const double c = c_x + c_y + c_z;
+
+    /*
+     * Find the determinant of the quadratic to ensure that it is not an
+     * imaginary solution.
+     */
+    const double determinant = pow(b, 2) - 4 * a * c;
+    if (determinant < 0)
+    {
+        ROS_INFO_STREAM("dX = " << delta_x);
+        ROS_INFO_STREAM("dY = " << delta_y);
+        ROS_INFO_STREAM("dZ = " << delta_z);
+        ROS_INFO_STREAM("a: " << a);
+        ROS_INFO_STREAM("b: " << b);
+        ROS_INFO_STREAM("c: " << c);
+        ROS_INFO_STREAM("Determinant: " << determinant);
```

Comment:
If this information is useful to include with the error when diagnosing issues, it would be better to have it labeled as a part of the error so that anyone running the node in future who is unfamiliar with why they are important is more likely to include them with a bug report.